core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -572,6 +573,9 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                 } else {
                     logger.debug("Skipping copying segments, current read-offset:{}, and LSO:{}", copiedOffset, lso);
                 }
+            } catch (CustomMetadataSizeLimitExceededException e) {
+                // Only stop this task. Logging is done where the exception is thrown.
+                this.cancel();
```

Comment:
I think we should update failed metrics as L584/585 did below.

Reply:
Yes, good point. I also added metrics check in `testCustomMetadataSizeExceedsLimit`.